src/sidebar/local-storage.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+/**
+ * Fallback in-memory store if `localStorage` is not read/writable.
+ */
+class InMemoryStorage {
+  constructor() {
+    this._store = {};
+  }
+
+  getItem(key) {
+    return (key in this._store) ? this._store[key] : null;
+  }
+
+  setItem(key, value) {
+    this._store[key] = value;
+  }
+
+  removeItem(key) {
+    delete this._store[key];
+  }
+}
```

Comment:
Nice. I can see that the CoffeeScript already contained the fallback in-memory behaviour with these same methods, but I appreciate seeing it in a named and docstring-ed class now, I think that's much clearer.